src/liboslexec/constfold.cpp

```diff
@@ -2444,11 +2452,16 @@ DECLFOLDER(constfold_gettextureinfo)
             newop.argwrite (1, false);
             return 1;
         } else {
-            // Return without constant folding gettextureinfo -- because
-            // we WANT the shader to fail and issue error messages at
-            // the appropriate time.
-            (void) rop.texturesys()->geterror (); // eat the error
-            return 0;
+            // Constant fold to 0
```

Comment:
@lgritz This case seems to be broken, but I am not sure why. Can you spot any obvious mistakes in how I'm inserting the call to "error"? The output of the unit test suggests it isn't getting called at all.

Reply:
I'm not sure yet, but Ole just got me a test scene on Friday, that I'm looking into today, as a test case for deep dive into exactly why some of those don't get optimized away. Probably going to kick myself when I figure it out.

Reply:
The shaders are using the idiom:

`gettextureinfo(filename, "channels", nchannel) == 0` as a way to detect if the file is valid.

Folding away this case seems worthwhile (although it isn't clear how often that case is actually triggered - since we never see an error message from it).



I also fixed the array case (did not require extra code), which should catch queries to `resolution` and `datawindow`, `displaywindow`. I'll push that in a second.

Reply:
Got it all working now.

The string in the testsuite output has to change, but that seems acceptable.

There are several cases where `gettextureinfo` fails but does not generate an error message and those cases can be folded without having to add a message into the opcode stream.

Reply:
Ah, that makes sense.